### PR TITLE
Refactor docs-content-link-rewrites/rewrite-script.ts

### PR DESCRIPTION
**Pull request by ashleemboyer**
## 🗒️ What/Why

- Updates `scripts/docs-content-link-rewrites/rewrite-links-plugin.ts` to leverage logic from our two server-side link rewrite plugins:
  - `remark-plugin-adjust-link-urls`
  - `rewrite-tutorial-links`
- Simplifies how "statistics" are tracked for the script's runs
  - There's now one conditional that checks if a link gets rewritten, and it's added to the appropriate data structure under the `statistics` argument 
- Removes now unneeded helpers:
  - `scripts/docs-content-link-rewrites/helpers/get-dot-io-to-dev-dot-url-map.ts`
  - `scripts/docs-content-link-rewrites/helpers/get-learn-to-dev-dot-url-map.ts`
- Shortens some of the less important `console.log`s so they don't take up as many lines in a terminal
- In `get-rewrite-links-script-arguments.ts`:
  - Removes unused `addPrefixToFilePaths`
  - Replaces the `contentDirectory` and `navDataDirectory` arguments with a `localCopyLocation` argument for easier CLI use
- Removes commented-out portions of the script that were intended for handling nav data (but leaves the `TODO` in case we come back to this part via this script)

## 🧪 Testing

We've been testing this script on several draft migration PRs:

- [vagrant#13048](https://github.com/hashicorp/vagrant/pull/13048)
- [boundary#2777](https://github.com/hashicorp/boundary/pull/2777)
- [packer#12192](https://github.com/hashicorp/packer/pull/12192)
- [nomad#15779](https://github.com/hashicorp/nomad/pull/15779)
- [consul#15976](https://github.com/hashicorp/consul/pull/15976)
- [vault#18696](https://github.com/hashicorp/vault/pull/18696)

### CODE REVIEW

File: scripts/docs-content-link-rewrites/rewrite-links-plugin.ts [hunk 1]
```diff
@@ -1,245 +1,58 @@
...
-interface VFileData {
-	linksToRewrite: Record<string, string>
-	unrewriteableLinks: string[]
-}
-
-interface CustomVFile extends VFile {
-	data: VFileData
-}
-
-const getIsRelativeUrl = (url: string) => {
```
Comment (kendallstrautman) [inline scripts/docs-content-link-rewrites/rewrite-links-plugin.ts:19]: ❓ Trying to keep up with all the refactors 😅 Was this code thats being deleted refactored into the shared plugin? I remember reviewing test cases that seem similar to what these functions did. 🔪 🔪

Comment (ashleemboyer) [reply scripts/docs-content-link-rewrites/rewrite-links-plugin.ts:19]: No worries! This code was totally deleted in favor of already existing code that lives in the two link rewriter plugins we use server-side when processing docs and tutorials content.

Comment (kendallstrautman) [reply scripts/docs-content-link-rewrites/rewrite-links-plugin.ts:19]: Okay that makes sense. So this script essentially plays off of the refactor where you wanted to leverage the existing rewrite plugins, accepting a generic `urlAdjustFn` – which is presumably where the tutorial / devdot url mapping work occurs? And those functions passed in are the shared ones with the tutorials / docs rewrite plugins?

Comment (kendallstrautman) [reply scripts/docs-content-link-rewrites/rewrite-links-plugin.ts:19]: Oh wait no I see the `processDocsNode` and the tutorial functions down below

Comment (kendallstrautman) [reply scripts/docs-content-link-rewrites/rewrite-links-plugin.ts:19]: Okay so this just refactors to use those same functions which we run in the tutorials / docs plugins.

Comment (ashleemboyer) [reply scripts/docs-content-link-rewrites/rewrite-links-plugin.ts:19]: Here's the recent PR where I abstracted the 2 plugin bodies into reusable functions (which I'm calling here in this refactor): https://github.com/hashicorp/dev-portal/pull/1530

**Review by kendallstrautman — APPROVED** [review]

**Pull request merged**